src/hooks/useStaking.js

```diff
@@ -38,12 +38,12 @@ export function useStaking() {
     stakingFactoryAbi
   )
 
-  const stakingContract = useContract(
+  const stakingContract = useContractReadOnly(
```

Comment:
This is to prevent errors on cases where the connected garden network is different from the connected wallet network.

And since we have the Multimodal implementation, there´s no need to have the full contract with signer.